### PR TITLE
Add Authentication to Portal/buckets/ APIs

### PR DESCRIPTION
**Pull request by cesnietor**
Added Login Api which creates a JWT token signed with a JWTkey from a Kubernetes Secret
The api `/api/login/` returns a Token that is required so far for all `/api/buckets/` requests
```
/login/ :
	Method: POST
	Description: Allows authentication and returns a Token
	Body payload request: {"username": "<user_name>", "password": "<pswd>", "tenant": <tenant_name>}
	Returns: {"token":"<jwt_token>"}
```
TODO: user DB is hardcoded so we need to add  it to get the proper user and passwords

### CODE REVIEW

File: cluster/setup.go [hunk 1]
```diff
@@ -68,9 +70,37 @@ func SetupM3Namespace() {
...
+		fmt.Println(err)
+		panic(err.Error())
+	}
+	// Create secret for JWT key for rest api
+	secret := v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "jwtkey",
+		},
+		Data: map[string][]byte{
+			"M3_JWT_KEY": []byte("jwtSecretKey"),
```
Comment (dvaldivia) [inline cluster/setup.go]: Can you generate a random string at least 64 characters long?

File: portal/api-authentication.go [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+	jwtgo "github.com/dgrijalva/jwt-go"
+	jwtreq "github.com/dgrijalva/jwt-go/request"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+// defaultJWTExpTime is the Portal expiration time for the authentication token.
+// Default is five minutes
+const defaultJWTExpTime = 5 * time.Minute
```
Comment (dvaldivia) [inline portal/api-authentication.go]: maybe make it a little longer, like 12 hours

File: portal/api-authentication.go [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+		},
+		"daniel@acme.com": {
+			"password": "daniel_hashed",
+			"uuid":     "00112233-4455-6677-8899-aabbccddeeff",
+		},
+	},
+}
+
+// TODO: jwtKey should be gotten from a m3 global config
+var jwtKey = []byte("secret_key")
```
Comment (dvaldivia) [inline portal/api-authentication.go]: I thought you were reading this from secrets

**Review by dvaldivia — CHANGES_REQUESTED** [review]

File: portal/api-authentication.go (hunk 3 above)
Comment (cesnietor) [reply portal/api-authentication.go]: yes, is not being used, already removed

**Comment by cesnietor** [comment]
Closing since it will be migrated to a grpc API

**Comment by cesnietor** [comment]
Closing since this will be migrated to a grpc API.

**Pull request closed without merging**